Update common.js

glennfrutiz opened the pull request:
I'm pushing just this function for now so Melissa can keep working on her part. 

Still updating my other features based on this morning's meeting.

I wasn't confident on using UNION in my query after this mornings meeting. Not sure how I can use it for this purpose since I don't think the columns would stay the same between the two queries I'm making.  I was also worried I wouldn't be able to tell apart userIds when the user is a requestor and when the user is a seller.

glennfrutiz committed:
Update common.js

jung8 reviewed (changes requested):
Here are my suggestions:
1) I think you have the same issue that I had this morning with async functions. The two queries execute independently of each other and your return call will execute before the queries finish. You need to implement promises or use a callback function as Melissa suggested this morning. I think the callback might be an easier option.

2) If you want to get the sum of things from SQL you can use the SUM() function in the query. For your first query, you could do:
SELECT SUM(pointCost) FROM Transactions WHERE = ? AND statusId <> 8;

I also think that you may need to adjust the query to check for more than just the complete status. I think it might be easier to just add a pending points column to the TRansactions table.

mkoeplin commented on common.js:
```diff
@@ -18,7 +19,54 @@ function isAuthenticated(req, res, next) {
+    FROM Transactions
+    INNER JOIN UserBooks on UserBooks.id = Transactions.userBookId
+    WHERE UserBooks.userId = ? AND Transactions.statusId <> 8`;
+
```
Agree with Andrews points here where you will need to use callback or promises but callback does seem easier and it's how I've done some of my transactions now so just for consistency we could go with that.

Also agree with Andrew you could just do a sum in each select statement and then you can just sum the two result numbers together.

Your query is correct though. @jung8 Did you misread his queries? The '<>' means not equal so he's all transactions that are not closed which is more efficient than doing an equals or in statement on the other 7 statuses.

mkoeplin replied:
The pending point column was the initial suggestion but since we decided to go this route and I have code that is already set up to use this I would ask that we don't suddenly change course as it will require me to do another refactor. This approach should be fine provided it's switched to use callback.

jung8 replied:
@mkoeplin I think that if we select all statuses that are not equal to 8 then you select all statuses 1-7. This is not what we want for pending points because this includes received, rejected, canceled, and lost.

mkoeplin reviewed (changes requested):
See comments in file.

glennfrutiz committed:
Update common.js

i think this is working? I am getting back a value for total points based on what I have in transactions. 

I am unsure if a value will be available to you when you call this though. If you still think this requires a call back function please let me know. I think i got it working with a callback function as well but i opted for this as it is much cleaner.

glennfrutiz committed:
Update common.js

removed type

mkoeplin reviewed (approved):
Nice job! Using one sql query makes this function super straight forward, I like it!

jung8 reviewed (approved):
I think the return part still needs to be updated to return the callback function to be visible outside the SQL query function.
